components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java

```diff
@@ -216,6 +207,27 @@ public static String jsFunction_getAuthServerURL(Context cx, Scriptable thisObj,
         return url;
     }
 
+    /**
+     * Get the identity provider URL from app-manager.xml file
+     *
+     * @param context Rhino context
+     * @param thisObj Scriptable object
+     * @param args    Passing arguments
+     * @param funObj  Function object
+     * @return identity provider URL
+     * @throws org.wso2.carbon.appmgt.api.AppManagementException Wrapped exception by org.wso2.carbon.apimgt.api.AppManagementException
+     */
+    public static String jsFunction_getIdentityProviderUrl(Context context, Scriptable thisObj,
+                                                           Object[] args,
+                                                           Function funObj) throws AppManagementException {
+        AppManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
+        String url = config.getFirstProperty(AppMConstants.SSO_CONFIGURATION_IDENTITY_PROVIDER_URL);
+        if (url == null) {
+            handleException("Identity provider URL unspecified");
```

Comment:
It is better to state the error message with more details. e.g. "Identity provider URL unspecified in <x> element in appmanager.xml" so that one who encounter the error can easily fix.